### ISSUE TITLE
Can multiline messages still show a timestamp on each line in the log?

### ISSUE THREAD

**Issue by nealmt** [weechat/weechat#2210]
### Feature description

I have a question about multiline and logging. (For context, I am using wee-slack.) Here's how messages look in my buffer:

```
22:19 neal | multi
           | line
           | message
22:26 neal | SNORK
           | MAGGLE
           | PLUM
```

Here's how I would like it to look in the log:

```
[2024-11-12 22:19:52]   neal    multi
[2024-11-12 22:19:52]           line
[2024-11-12 22:19:52]           message
[2024-11-12 22:26:57]   neal    SNORK
[2024-11-12 22:26:57]           MAGGLE
[2024-11-12 22:26:57]           PLUM
```

But here's how it appears in the log:

```
[2024-11-12 22:19:52]   neal    multi
line
message
[2024-11-12 22:26:57]   neal    SNORK
MAGGLE
PLUM
```

Is it possible to allow logs to still have a timestamp for each line? I appreciate any insight! Thank you.

**Comment by flashcode** [comment]
Hi,

This is currently not possible to do that, the log would not reflect exactly what is displayed by WeeChat, but that should be possible to do that with an option.

The problem is the parsing of the logs in case of multiline messages?

**Comment by flashcode** [comment]
I think doing that would also break the backlog displayed when the log file is loaded (on new buffer): the user messages will be split into multiple messages, and only the first one will be linked to the nick.

**Comment by nealmt** [comment]
Hi, thank you for the reply, @flashcode!

> The problem is the parsing of the logs in case of multiline messages?

Yes, that's it. Well, kind of. The log correctly shows the same information that was displayed in the buffer, but it would be helpful in my case to show the multiple lines in the log along with their associated timestamps.

> I think doing that would also break the backlog displayed when the log file is loaded (on new buffer): the user messages will be split into multiple messages, and only the first one will be linked to the nick.

I don't have any idea about that, unfortunately, since I have only (so far) used WeeChat with wee-slack, and I don't think it runs into this issue. (Correct me if I'm wrong.)

**Comment by trygveaa** [comment]
> I have only (so far) used WeeChat with wee-slack, and I don't think it runs into this issue. (Correct me if I'm wrong.)

That's correct. wee-slack gets backlog from the Slack API instead of using the log files. But it would be an issue e.g. with IRC servers that support multiline.

**Comment by flashcode** [comment]
Maybe the default behavior could be to split messages anyway by default, as this makes logs easier to parse, and there would just be a minor change when backlog is displayed: multiple messages instead of a single one.
